src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -449,6 +449,23 @@ def construct_payload(self):
             safe_set(self.new_containerapp, "properties", "template", "revisionSuffix", value=None)
 
         if self.get_argument_revisions_mode():
+            if self.get_argument_revisions_mode().lower() == "labels":
+                # Transitioning into labels mode is complicated and we don't want to combine it with other updates.
+                # Check if the app was previously in labels mode. If not, throw an error saying to use Set-Mode instead.
+                containerapp_def = None
+                try:
+                    containerapp_def = ContainerAppPreviewClient.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+                except Exception as e:
+                    handle_raw_exception(e)
+
+                if not containerapp_def:
+                    raise ResourceNotFoundError("The containerapp '{}' does not exist".format(self.get_argument_name()))
+
+                if safe_get(containerapp_def, "properties", "configuration", "activeRevisionsMode").lower() != "labels":
+                    raise ArgumentUsageError(
+                        "The containerapp '{}' is not in labels mode. Please use `az containerapp revision set-mode` to switch to labels mode first.".format(
```

Comment:
later we can remove "labels" value for "containerapp update" and "up" command.

Reply:
Good point, I had added activeRevisionsMode to up and update before I knew about set-mode, but we could remove them. The tricky part for up is that we do want to be able to create an app in labels mode.